File: Reverse Bits/Reverse_Bits.py
# Runtime: 24 ms, faster than 69.61% of Python online submissions for Reverse Bits.
# Memory Usage: 11.9 MB, less than 5.38% of Python online submissions for Reverse Bits.

class Solution:
    # @param n, an integer
    # @return an integer
    def reverseBits(self, n):
        res = 0
        mask = 1

        for i in range(32):
            res <<= 1
            if n & mask:
                res += 1
            mask <<= 1
        
        return res

File: Reverse Bits/Reverse_Bits.cpp
// Runtime: 4 ms, faster than 100.00% of C++ online submissions for Reverse Bits.
// Memory Usage: 8.3 MB, less than 5.36% of C++ online submissions for Reverse Bits.

class Solution 
{
public:
    uint32_t reverseBits(uint32_t n) 
    {
        uint32_t res = 0, mask = 1;
        for (int i = 0; i < 32; i++)
        {
            res <<= 1;
            if (n & mask)
                res += 1;
            mask <<= 1;
        }
        return res;
    }
};